elastic/kibana
# [App Search] New Add Domain Flyout for Crawler

## Summary

Migrates the existing `AddDomainModal` from `ent-search` into a new `AddDomainFlyout` component.  I'll be adding https/https validation and our 4-step visual validation process in follow-up PRs.  

## Screenshots

<img width="800" alt="Screen Shot 2021-07-19 at 7 35 08 PM" src="https://user-images.githubusercontent.com/2479295/126240786-818c1968-2ad4-4cb1-afaa-6733a13d0088.png">

<img width="800" alt="Screen Shot 2021-07-19 at 7 36 00 PM" src="https://user-images.githubusercontent.com/2479295/126240793-fe07a44f-e337-47e9-a953-26c685c66bb9.png">

<img width="800" alt="Screen Shot 2021-07-19 at 7 36 05 PM" src="https://user-images.githubusercontent.com/2479295/126240800-05a471bf-307b-4360-a26b-e16fedc36766.png">


## QA

1. Create a new non-meta engine and upload some documents, or use and existing non-meta engine
2. Navigate to the Crawler Overview for that engine
 - [ ] There is an "Add Domain" button above the domains table
3. Click the "Add Domain" button
- [ ] A flyout appears containing a form to add a new domain
- [ ] The submit button is initially disabled
- [ ] There is already "https://" entered for the user in the input box
- [ ] There is a button to validate the contents of the input box
4. Enter a domain to add.  (I use `https://elastic.co` and `https://elastic.co/guide` 99% of the time)
5. Click the button to validate your domain
- [ ] The submit button at the button of the flyout should now be enabled
6. Click the button to submit your domain
- [ ] You should be navigated to the (not implemented yet) single domain view for your domain
- [ ] A success toast should appear

### Checklist

Delete any items that are not applicable to this PR.
